Minor refactor on views.py on newsletter app.

niwinz opened the pull request:
Esto, no esta probado, es decir, se que minimamente no la he cagado pero no lo he probado bien.

Por otro lado me gustaria ver, por que se usa password como token... luego, el username de django es muy pequeño para colocar los emails, tengo que ver una solucion para ello.

He normalizado el codigo para que sea mas limpio, he puesto cadenas unicode en todos los lados que no estaban puestos y he eliminado niveles de indentacion absurdos :D 

Opiniones, reviews!

Commit:
Minor refactor on views.py on newsletter app.

achamorro-dev commented:
Niveles de indentación absurdos mea culpa! Mezcle configuraciones de vim y
se me olvido corregir las vistas :)

A lo de unicode me tengo que acostumbrar si. El email como nombre de
usuario se puede eliminar, simplemente lo hice por añadir algo como
usuario. Habría que quitarlo de la validación posterior también. Lo que
tiene hacer las cosas deprisa y corriendo :(

Utilizar la password como token es para dar un enlace único para darse de
baja. Si usamos solo el email yo podría ir probando y eliminar gente que no
soy yo :) ¿Alguna idea mejor? Siempre son bienvenidas.

Podria dar merge yo porque no lo veo mal pero soy un poco novato en estos
tramites y prefiero que revisen otros jajaja

El 5 de marzo de 2013 12:22, Andrey Antukh notifications@github.comescribió:

> Esto, no esta probado, es decir, se que minimamente no la he cagado pero
> no lo he probado bien.
> 
> Por otro lado me gustaria ver, por que se usa password como token...
> luego, el username de django es muy pequeño para colocar los emails, tengo
> que ver una solucion para ello.
> 
> He normalizado el codigo para que sea mas limpio, he puesto cadenas
> unicode en todos los lados que no estaban puestos y he eliminado niveles de
> indentacion absurdos :D
> 
> ## Opiniones, reviews!
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/python-spain/pycones-web bugfix/newsletter-views
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/python-spain/pycones-web/pull/6
> Commit Summary
> - Minor refactor on views.py on newsletter app.
> 
> File Changes
> - _M_ pycones/newsletter/views.pyhttps://github.com/python-spain/pycones-web/pull/6/files#diff-0(103)
> 
> Patch Links:
> - https://github.com/python-spain/pycones-web/pull/6.patch
> - https://github.com/python-spain/pycones-web/pull/6.diff

niwinz commented:
No mergeen de momento, por que quiero review. Respeto a token y username/email lo reviso esta tarde y añadire mas commits a este pull request.

Respeto al token. Lo ideal es mantener otro campo, lo voy a hacer yo en cuanto pueda.

Commit:
Newsletter model refactor and improved queries.

Commit:
Fix typos.

niwinz commented:
Acabo de subir otro commit, obviamente sin probar y no se os ocurra mergear.
Reviews bienvenidos.

He modificado el modelo de newsletter para que sea mas dummy y he hecho que demomento solo un newsletter puede estar activo (por settings) y todas las queries en el manager automaticamente se filtran por el newsletter activo..

A igual que todo esto, cuando haces la subscripcion de subscribe a un newsletter activo actual.
Reviews bienvenidas.

ghost commented:
> ... el username de django es muy pequeño para colocar los emails, tengo que ver una solucion para ello.

Es a lo único a lo que le sacaría una pega. Podríamos usar un hash del email.

Estaba pensando en escribir unas pruebas para comprobar que un usuario puede suscribirse y desuscribirse. ¿@niwibe lo estás haciendo? Para no duplicar esfuerzos.

niwinz commented:
No estoy escribiendo las pruebas, pero si estoy refactorizando esta parte, puedes hacer los tests unitarios que ataquen las urls si quieres, despues de mi refactor no deberian cambiar mucho esos tests :D

Commit:
Refactor welcome email-send function.

Commit:
Add missing __init__.py file on profile module.

Commit:
Fix typo on models.py

Commit:
Test de suscripción al newsletter

Verifica que el formulario de suscripción devuelve HTTP 200 cuando se da la
dirección de correo electrónico.

He añadido `django-nose` como dependencia para desarrollar. Hace más sencillo
ejecutar los tests y autodescubre tests en clases y funciones.

Commit:
Add local_settings.py.sample default file for development environments.

Commit:
Fixed email template encoding.

Commit:
Minor changes on urls.py on newsletter app.

Commit:
Make subscribe view transactional.

Commit:
Fix Email subclass import.

Commit:
Add missing Profile import and other minor name changes.

niwinz commented:
Bueno, parece que ya lo tengo terminado y probado que funciona razobablemente bien.
Si Alejandro se anima a hacer unos tests unitarios todos estariamos muy agradecidos.

Por favor necesito un review y ack de esto de alguna persona para que lo acabe mergeando.
Despues de esto me pondre con el tema de el tamaño de username/email :D con una tecnica que ya he aplicado en otro lado.

Commit:
No cambiar al test runner de nose si DEBUG es False

Commit:
Test de desuscripción del newsletter

ghost commented:
He escrito un par de tests básicos para comprobar que la suscripción y desuscripción
funciona correctamente. Parece que sin problemas :green_heart: 

> Por favor necesito un review y ack de esto de alguna persona para que lo acabe mergeando.

Voy a dejar que lo mergee otra persona ya que he subido commits también.

niwinz commented:
Ya lo mergeo yo. Solo queria que lo vieran otros. Mañana lo mergeo.

Sent from my Nexus 4
On Mar 6, 2013 12:24 AM, "Alejandro Gómez" notifications@github.com wrote:

> He escrito un par de tests básicos para comprobar que la suscripción y
> desuscripción
> funciona correctamente. Parece que sin problemas [image: :green_heart:]
> 
> Por favor necesito un review y ack de esto de alguna persona para que lo
> acabe mergeando.
> 
> Voy a dejar que lo mergee otra persona ya que he subido commits también.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/python-spain/pycones-web/pull/6#issuecomment-14472540
> .

ghost commented:
:+1:

niwinz commented:
Merged!
